pkg/ansible/watches/watches.go

```diff
@@ -149,50 +149,76 @@ func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
+			}
+		}
 	}
 	if w.Finalizer != nil && len(w.Finalizer.Playbook) > 0 {
 		w.Finalizer.Playbook = getFullPath(w.Finalizer.Playbook)
 	}
 }
 
-// getFullPath will return a valid full path for the playbook
+// getFullPath returns an absolute path for the playbook
 func getFullPath(path string) string {
 	if len(path) > 0 && !filepath.IsAbs(path) {
 		return filepath.Join(projutil.MustGetwd(), path)
 	}
 	return path
 }
 
-// getFullRolePath will return a valid full path for the role
-func getFullRolePath(path string) string {
-	if len(path) > 0 && !filepath.IsAbs(path) {
-		envVar, ok := os.LookupEnv("ANSIBLE_ROLES_PATH")
-		if ok && len(envVar) > 0 {
-			// Check all values informed (e.g. path1/roles:path2/roles:/absolute/path3/roles)
-			result := strings.Split(envVar, ":")
-			for i := range result {
-				// Check if the role can be found in the envVar + role path
-				infoPath := filepath.Join(result[i], path)
-				if _, err := os.Stat(infoPath); err == nil {
-					return infoPath
-				}
-
-				// Check if the role can be found in the envVar + roles + role path
-				infoPathWithRolesDir := filepath.Join(result[i], "roles", path)
-				if _, err := os.Stat(infoPathWithRolesDir); err == nil {
-					return infoPathWithRolesDir
-				}
+// getFullRolePath returns a full path to the role directory
+func getPossibleRolePaths(path string) []string {
+	possibleRolePaths := []string{}
+	if filepath.IsAbs(path) || len(path) == 0 {
+		return append(possibleRolePaths, path)
+	}
+	fqcn := strings.Split(path, ".")
+	// If fqcn is a valid fully qualified collection name, it is <namespace>.<collectionName>.<roleName>
+	if len(fqcn) == 3 {
+		ansibleCollectionsPathEnv, ok := os.LookupEnv("ANSIBLE_COLLECTIONS_PATH")
+		if !ok || len(ansibleCollectionsPathEnv) == 0 {
+			ansibleCollectionsPathEnv = "/usr/share/ansible/collections"
+			home, err := os.UserHomeDir()
+			if err == nil {
+				homeCollections := filepath.Join(home, ".ansible/collections")
+				ansibleCollectionsPathEnv = ansibleCollectionsPathEnv + ":" + homeCollections
 			}
 		}
-		// If the flag and/or env var ANSIBLE_ROLES_PATH was not set or no roles were informed in the path then, it will
-		// be the current directory concat with roles.
-		return getFullPath(filepath.Join("roles", path))
+		for _, possiblePathParent := range strings.Split(ansibleCollectionsPathEnv, ":") {
+			possibleRolePaths = append(possibleRolePaths, filepath.Join(possiblePathParent, "ansible_collections", fqcn[0], fqcn[1], "roles", fqcn[2]))
+		}
 	}
-	return path
+
+	// Check for the role where Ansible would. If it exists, use it.
+	ansibleRolesPathEnv, ok := os.LookupEnv("ANSIBLE_ROLES_PATH")
+	if ok && len(ansibleRolesPathEnv) > 0 {
+		for _, possiblePathParent := range strings.Split(ansibleRolesPathEnv, ":") {
+			// "roles" is optionally a part of the path. Check with, and without.
+			// TODO(asmacdo) ^This might not be true. https://github.com/operator-framework/operator-sdk/pull/2273#discussion_r374946618
```

Comment:
The Ansible documentation do not explain how it should work. 
However, according to it maintainer

> A role should be found in the defined "roles path" (via config) or in a roles/ directory adjacent to the play
From: https://github.com/ansible/ansible/issues/16911#issuecomment-236765123

Because of this, we are using roles too.